akka-remote/src/main/scala/akka/remote/artery/LruBoundedCache.scala

```diff
@@ -0,0 +1,234 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.util.{ OptionVal, Unsafe }
+
+import scala.annotation.tailrec
+import scala.reflect.ClassTag
+
+object FastHash {
+
+  val Prime1: Int = 32717
+  val Prime2: Int = 32749
+  val Prime3: Int = 65519
+  val Prime4: Int = 65521
+
+  // FIXME: !!! Totally unscientific hash !!!, until I figure out how to port xxHash efficiently (it needs 64 bit access to be
+  // fast so we need the ByteBuffer instead of the String)
+  def ofString(s: String): Int = {
```

Comment:
It is too weak. As for caching, we only access the hashcode once for a String instance (they are freshly created by the header parsing code) so I am not sure it helps. We can use String.hashCode though as a temporary measure. What I am actually after is to have the header parser calculate hashes directly from the underlying ByteBuffer, as it allows to read 64 bits (8 characters) directly. Then both the compression and the cache could just use that returned hash value (values).